GeneralsMD/Code/GameEngine/Source/GameClient/SelectionInfo.cpp

```diff
@@ -385,12 +385,6 @@ Bool addDrawableToList( Drawable *draw, void *userData )
 			{
 				return FALSE;
 			}
-
-			//If stealthed, no way!
-			if( obj->testStatus( OBJECT_STATUS_STEALTHED ) && !obj->testStatus( OBJECT_STATUS_DETECTED ) )
-			{
-				return FALSE;
```

Comment:
With this condition gone, does it mean that we can now select enemy stealth units, while they are stealth? Or is that covered by `isDrawableEffectivelyHidden` ?

If so, then this fix needs to be different, for example allowing the observer to skip these tests.

Reply:
Absolutely not, or this change would be a regression. The original stealth selection condition is now covered by `isDrawableEffectivelyHidden`. The benefit to this is that as observers can see stealthed units and beacons, `isDrawableEffectivelyHidden` returns false for them (unless the units are in a closed container).

https://github.com/user-attachments/assets/748b188b-6e9b-477f-8d07-697ee051fa89

https://github.com/user-attachments/assets/ecd79c6a-b564-41d9-b83d-be2d71f73d8c